File: src/components/AllReflectionsPage.js
import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useTheme } from '../contexts/ThemeContext';
import { ArrowLeft, Sparkles, Check, Calendar, X, ChevronLeft, ChevronRight } from 'lucide-react';
import { getCurrentUser } from '../services/authService';
import firestoreService from '../services/firestoreService';
import { formatDateForDisplay, getDateId } from '../utils/dateUtils';
import CalendarPopup from './CalendarPopup';

export default function AllReflectionsPage() {
  const navigate = useNavigate();
  const { isDarkMode } = useTheme();
  const [reflections, setReflections] = useState([]);
  const [filteredReflections, setFilteredReflections] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  const [selectedDate, setSelectedDate] = useState(null);
  const [isCalendarOpen, setIsCalendarOpen] = useState(false);
  const [reflectionDays, setReflectionDays] = useState([]);

  // Format date for display (e.g., "8 October 2025 • Wed, 3:54 pm")
  const formatReflectionDate = (reflectionItem) => {
    if (!reflectionItem) return '';
    try {
      // Use createdAt if available (has actual time), otherwise use dateObj or date
      let date = reflectionItem.createdAt || reflectionItem.dateObj;
      
      if (!date) {
        // Fallback to parsing date string
        const dateString = reflectionItem.date;
        if (dateString && dateString.includes('-')) {
          const [year, month, day] = dateString.split('-');
          date = new Date(parseInt(year), parseInt(month) - 1, parseInt(day));
        } else if (dateString) {
          date = new Date(dateString);
        } else {
          return '';
        }
      }
      
      // Handle Firestore Timestamp
      if (date && typeof date.toDate === 'function') {
        date = date.toDate();
        }
      
      if (!date || isNaN(date.getTime())) {
        return reflectionItem.date || '';
      }
      
      const day = date.getDate();
      const month = date.toLocaleDateString('en-US', { month: 'long' });
      const year = date.getFullYear();
      const dayName = date.toLocaleDateString('en-US', { weekday: 'short' });
      const time = date.toLocaleTimeString('en-US', { 
        hour: 'numeric', 
        minute: '2-digit',
        hour12: true 
      });
      
      return `${day} ${month} ${year} • ${dayName}, ${time}`;
    } catch (error) {
      console.error('Error formatting date:', error);
      return reflectionItem.date || '';
    }
  };

  // Load all reflections
  useEffect(() => {
    const loadReflections = async () => {
      const user = getCurrentUser();
      if (!user) {
        setIsLoading(false);
        return;
      }

      try {
        setIsLoading(true);
        
        const allReflections = [];
        
        // Get all pod reflections from podReflections collection
        const podReflectionsResult = await firestoreService.getAllPodReflections(user.uid);
        if (podReflectionsResult.success && podReflectionsResult.reflections) {
          podReflectionsResult.reflections.forEach(ref => {
            let reflectionDate;
            try {
              if (ref.dateId && ref.dateId.includes('-')) {
                const [year, month, day] = ref.dateId.split('-');
                reflectionDate = new Date(parseInt(year), parseInt(month) - 1, parseInt(day));
              } else {
                reflectionDate = ref.createdAt || new Date();
              }
            } catch (e) {
              reflectionDate = ref.createdAt || new Date();
            }
            
            allReflections.push({
              id: ref.id,
              date: ref.dateId || (ref.createdAt ? getDateId(ref.createdAt) : getDateId(new Date())),
              dateObj: reflectionDate,
              reflection: ref.reflection,
              createdAt: ref.createdAt || reflectionDate
            });
          });
        }
        
        // Also get all pods with reflections (as backup/alternative source)
        const result = await firestoreService.getAllPods(user.uid);
        if (result.success && result.pods) {
          // Process pods with reflections
          result.pods
            .filter(pod => pod.reflection && pod.startDate)
            .forEach(pod => {
              // Check if this reflection already exists from podReflections
              const exists = allReflections.some(r => r.date === pod.startDate);
              if (!exists) {
              // Create date object from startDate
              let reflectionDate;
              try {
                if (pod.startDate.includes('-')) {
                  const [year, month, day] = pod.startDate.split('-');
                  reflectionDate = new Date(parseInt(year), parseInt(month) - 1, parseInt(day));
                } else {
                  reflectionDate = new Date(pod.startDate);
                }
              } catch (e) {
                reflectionDate = new Date();
              }
                
                // Handle createdAt timestamp properly
                let createdAt = pod.createdAt;
                if (createdAt && typeof createdAt.toDate === 'function') {
                  createdAt = createdAt.toDate();
                } else if (!createdAt || !(createdAt instanceof Date)) {
                  createdAt = reflectionDate;
                }
              
              allReflections.push({
                id: pod.id || pod.startDate,
                date: pod.startDate,
                dateObj: reflectionDate,
                reflection: pod.reflection,
                  createdAt: createdAt
              });
              }
            });
        }
          
        // Also get current reflection if it exists and not already in list
          const currentReflectionResult = await firestoreService.getPodReflection(user.uid);
          if (currentReflectionResult.success && currentReflectionResult.reflection) {
            const today = new Date();
          const todayId = getDateId(today);
            
            // Check if today's reflection is already in the list
            const todayExists = allReflections.some(r => r.date === todayId);
            if (!todayExists) {
            let createdAt = currentReflectionResult.createdAt;
            if (createdAt && typeof createdAt.toDate === 'function') {
              createdAt = createdAt.toDate();
            } else if (!createdAt) {
              createdAt = today;
            }
            
              allReflections.push({
                id: 'current',
              date: currentReflectionResult.dateId || todayId,
                dateObj: today,
                reflection: currentReflectionResult.reflection,
              createdAt: createdAt
              });
          }
        }
        
        // Sort by date (newest first)
        allReflections.sort((a, b) => {
          const dateA = a.createdAt?.getTime() || a.dateObj?.getTime() || 0;
          const dateB = b.createdAt?.getTime() || b.dateObj?.getTime() || 0;
          return dateB - dateA;
        });
        
        // Extract reflection days for calendar
        const daysWithReflections = allReflections.map(r => ({
          date: r.date,
          hasReflection: true
        }));
        setReflectionDays(daysWithReflections);
        
        setReflections(allReflections);
        setFilteredReflections(allReflections);
      } catch (error) {
        console.error('Error loading reflections:', error);
      } finally {
        setIsLoading(false);
      }
    };

    loadReflections();
  }, []);

  // Filter reflections based on selected date
  useEffect(() => {
    if (selectedDate) {
      const dateId = getDateId(selectedDate);
      const filtered = reflections.filter(r => r.date === dateId);
      setFilteredReflections(filtered);
    } else {
      setFilteredReflections(reflections);
    }
  }, [selectedDate, reflections]);

  const handleCalendarClick = () => {
    setIsCalendarOpen(true);
  };

  const handleDateSelect = (date) => {
    setSelectedDate(date);
    setIsCalendarOpen(false);
  };

  const handleClearDateFilter = () => {
    setSelectedDate(null);
  };

  const getCurrentReflectionIndex = () => {
    if (!selectedDate || reflections.length === 0) return -1;
    const currentDateId = getDateId(selectedDate);
    return reflections.findIndex(r => r.date === currentDateId);
  };

  const handlePreviousDate = () => {
    const currentIndex = getCurrentReflectionIndex();
    if (currentIndex > 0) {
      const prevReflection = reflections[currentIndex - 1];
      if (prevReflection && prevReflection.dateObj) {
        setSelectedDate(prevReflection.dateObj);
      }
    }
  };

  const handleNextDate = () => {
    const currentIndex = getCurrentReflectionIndex();
    if (currentIndex >= 0 && currentIndex < reflections.length - 1) {
      const nextReflection = reflections[currentIndex + 1];
      if (nextReflection && nextReflection.dateObj) {
        setSelectedDate(nextReflection.dateObj);
      }
    }
  };

  const canGoPrevious = () => {
    const currentIndex = getCurrentReflectionIndex();
    return currentIndex > 0;
  };

  const canGoNext = () => {
    const currentIndex = getCurrentReflectionIndex();
    return currentIndex >= 0 && currentIndex < reflections.length - 1;
  };

  return (
    <div
      className="min-h-screen px-6 py-8 pb-20 relative overflow-hidden"
      style={{
        background: isDarkMode
          ? "#131313"
          : "#FAFAF8"
      }}
    >
      {/* Header */}
      <div className="mb-6">
        <div className="flex items-center space-x-3 mb-2">
          <button
            onClick={() => navigate('/pod')}
            className={`w-10 h-10 rounded-full flex items-center justify-center ${
              isDarkMode ? 'hover:bg-gray-800/50' : 'hover:bg-gray-100'
            } transition-colors`}
          >
            <ArrowLeft className={`w-5 h-5 ${isDarkMode ? 'text-white' : 'text-gray-800'}`} />
          </button>
          <div className="flex items-center space-x-2">
            <Sparkles className={`w-5 h-5 ${isDarkMode ? 'text-[#FDD663]' : 'text-[#E6B3BA]'}`} />
            <h1 className={`text-xl font-bold ${isDarkMode ? 'text-white' : 'text-gray-800'}`}>
              Reflections
            </h1>
          </div>
        </div>
        <p className={`text-sm ml-14 mb-3 ${isDarkMode ? 'text-gray-400' : 'text-gray-600'}`}>
          All your crew reflections
        </p>
        
        {/* Date Search */}
        <div className="ml-14 mb-4">
          <div
            onClick={handleCalendarClick}
            className={`rounded-lg px-3 py-2 flex items-center ${selectedDate ? 'justify-between' : 'justify-center'} cursor-pointer hover:opacity-80 transition-opacity ${
              isDarkMode ? 'backdrop-blur-md' : 'bg-white'
            }`}
            style={isDarkMode ? {
              backgroundColor: "#262626",
              boxShadow: "0 2px 8px rgba(0, 0, 0, 0.15)",
              border: "1px solid rgba(255, 255, 255, 0.08)",
            } : {
              boxShadow: "0 2px 6px rgba(0, 0, 0, 0.08)",
            }}
          >
            <div className={`flex items-center ${selectedDate ? 'space-x-2' : 'flex-col space-y-1'}`}>
              <Calendar className="w-4 h-4" style={{ color: isDarkMode ? "#7DD3C0" : "#87A96B" }} />
              <div className={selectedDate ? '' : 'text-center'}>
                <div className={`text-sm font-medium ${isDarkMode ? 'text-white' : 'text-gray-800'}`}>
                  {selectedDate ? formatDateForDisplay(selectedDate) : 'Search by date'}
                </div>
                <div className={`text-xs ${isDarkMode ? 'text-gray-400' : 'text-gray-500'}`}>
                  {selectedDate ? 'Tap to change date' : 'Tap to select a date'}
                </div>
              </div>
            </div>
            {selectedDate && (
              <button
                onClick={(e) => {
                  e.stopPropagation();
                  handleClearDateFilter();
                }}
                className="ml-2 p-1 rounded-full hover:opacity-80 transition-opacity"
              >
                <X className={`w-4 h-4 ${isDarkMode ? 'text-gray-400' : 'text-gray-500'}`} />
              </button>
            )}
          </div>
        </div>

        {/* Date Navigation (when date is selected) */}
        {selectedDate && filteredReflections.length > 0 && (
          <div className="ml-14 mb-4 flex items-center justify-center space-x-4">
            <button
              onClick={handlePreviousDate}
              disabled={!canGoPrevious()}
              className={`p-2 rounded-full transition-opacity ${
                canGoPrevious()
                  ? `hover:opacity-80 ${isDarkMode ? 'hover:bg-gray-800/50' : 'hover:bg-gray-100'}`
                  : 'opacity-30 cursor-not-allowed'
              }`}
            >
              <ChevronLeft className={`w-5 h-5 ${isDarkMode ? 'text-gray-400' : 'text-gray-600'}`} />
            </button>
            <div className={`text-base font-semibold ${isDarkMode ? 'text-white' : 'text-gray-800'}`}>
              {formatDateForDisplay(selectedDate)}
            </div>
            <button
              onClick={handleNextDate}
              disabled={!canGoNext()}
              className={`p-2 rounded-full transition-opacity ${
                canGoNext()
                  ? `hover:opacity-80 ${isDarkMode ? 'hover:bg-gray-800/50' : 'hover:bg-gray-100'}`
                  : 'opacity-30 cursor-not-allowed'
              }`}
            >
              <ChevronRight className={`w-5 h-5 ${isDarkMode ? 'text-gray-400' : 'text-gray-600'}`} />
            </button>
          </div>
        )}
      </div>

      {/* Reflections List */}
      {isLoading ? (
        <div className="flex flex-col items-center justify-center py-12">
          <div className="flex space-x-1 mb-3">
            <div className={`w-2 h-2 rounded-full animate-bounce ${isDarkMode ? 'bg-gray-400' : 'bg-gray-500'}`} style={{ animationDelay: '0ms' }}></div>
            <div className={`w-2 h-2 rounded-full animate-bounce ${isDarkMode ? 'bg-gray-400' : 'bg-gray-500'}`} style={{ animationDelay: '150ms' }}></div>
            <div className={`w-2 h-2 rounded-full animate-bounce ${isDarkMode ? 'bg-gray-400' : 'bg-gray-500'}`} style={{ animationDelay: '300ms' }}></div>
          </div>
          <p className={`text-sm ${isDarkMode ? 'text-gray-400' : 'text-gray-500'}`}>
            Loading reflections...
          </p>
        </div>
      ) : filteredReflections.length === 0 ? (
        <div className="flex flex-col items-center justify-center py-12">
          <div
            className={`w-16 h-16 rounded-lg flex items-center justify-center mb-4 ${
              isDarkMode ? 'backdrop-blur-md' : 'bg-white'
            }`}
            style={isDarkMode ? {
              backgroundColor: "rgba(28, 31, 46, 0.5)",
              boxShadow: "inset 0 0 20px rgba(125, 211, 192, 0.12), 0 8px 32px rgba(125, 211, 192, 0.08)",
              border: "1px solid rgba(125, 211, 192, 0.18)",
            } : {
              boxShadow: "0 2px 6px rgba(0, 0, 0, 0.06)",
            }}
          >
            <span className="text-3xl">🌿</span>
          </div>
          <p className={`text-sm text-center ${isDarkMode ? 'text-gray-400' : 'text-gray-600'}`}>
            {selectedDate ? 'No reflection found for the selected date.' : 'No reflections yet. Start chatting in your crew to generate reflections!'}
          </p>
        </div>
      ) : (
        <div className="space-y-3 max-w-sm mx-auto">
          {filteredReflections.map((reflection) => (
            <div
              key={reflection.id}
              className={`rounded-2xl p-4 relative overflow-hidden ${
                isDarkMode ? 'backdrop-blur-lg' : 'bg-white'
              }`}
              style={isDarkMode ? {
                backgroundColor: "#262626",
                boxShadow: "0 4px 16px rgba(0, 0, 0, 0.15)",
                border: "1px solid rgba(255, 255, 255, 0.08)",
              } : {
                boxShadow: "0 4px 12px rgba(0, 0, 0, 0.08)",
              }}
            >
              <div className="flex items-start gap-3">
                <div className="flex-shrink-0 mt-1">
                  <Check className={`w-4 h-4 ${isDarkMode ? 'text-[#81C995]' : 'text-[#87A96B]'}`} />
                </div>
                <div className="flex-1 min-w-0">
                  <div className={`text-xs font-medium mb-2 ${isDarkMode ? 'text-gray-400' : 'text-gray-500'}`}>
                    {formatReflectionDate(reflection)}
                  </div>
                  <p className={`text-sm leading-relaxed whitespace-pre-wrap break-words ${isDarkMode ? 'text-gray-300' : 'text-gray-700'}`}>
                    {reflection.reflection}
                  </p>
                </div>
              </div>
            </div>
          ))}
        </div>
      )}

      {/* Calendar Popup */}
      <CalendarPopup
        isOpen={isCalendarOpen}
        onClose={() => setIsCalendarOpen(false)}
        selectedDate={selectedDate || new Date()}
        onDateSelect={handleDateSelect}
        chatDays={reflectionDays}
      />
    </div>
  );
}

File: flutter_app/lib/screens/all_reflections_page.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:lucide_icons_flutter/lucide_icons.dart';
import 'package:provider/provider.dart';

import '../components/calendar_popup.dart';
import '../components/hub_theme.dart';
import '../components/hub_widgets.dart';
import '../contexts/theme_context.dart';
import '../router/app_router.dart';
import '../services/auth_service.dart';
import '../services/firestore_service.dart';
import '../services/firestore_result.dart';
import '../utils/date_utils.dart';

class _ReflectionItem {
  _ReflectionItem({
    required this.id,
    required this.date,
    required this.reflection,
    this.dateObj,
    this.createdAt,
  });

  final String id;
  final String date;
  final String reflection;
  final DateTime? dateObj;
  final DateTime? createdAt;
}

/// Mirrors src/components/AllReflectionsPage.js
class AllReflectionsPage extends StatefulWidget {
  const AllReflectionsPage({super.key});

  @override
  State<AllReflectionsPage> createState() => _AllReflectionsPageState();
}

class _AllReflectionsPageState extends State<AllReflectionsPage> {
  List<_ReflectionItem> _reflections = [];
  List<_ReflectionItem> _filtered = [];
  bool _loading = true;
  DateTime? _selectedDate;
  bool _calendarOpen = false;
  List<CalendarDayMarker> _reflectionDays = [];

  @override
  void initState() {
    super.initState();
    _load();
  }

  Future<void> _load() async {
    final user = AuthService().getCurrentUser();
    if (user == null) {
      setState(() => _loading = false);
      return;
    }
    setState(() => _loading = true);
    final all = <_ReflectionItem>[];

    final podRefs = await FirestoreService.instance.getAllPodReflections(user.uid);
    if (podRefs.success) {
      final refs = podRefs['reflections'] as List? ?? [];
      for (final raw in refs) {
        final ref = Map<String, dynamic>.from(raw as Map);
        DateTime? reflectionDate;
        final dateId = ref['dateId'] as String?;
        if (dateId != null && dateId.contains('-')) {
          final p = dateId.split('-');
          reflectionDate = DateTime(int.parse(p[0]), int.parse(p[1]), int.parse(p[2]));
        } else {
          reflectionDate = ref['createdAt'] is DateTime ? ref['createdAt'] as DateTime : DateTime.now();
        }
        all.add(_ReflectionItem(
          id: ref['id'] as String? ?? '',
          date: dateId ?? getDateId(reflectionDate),
          dateObj: reflectionDate,
          reflection: ref['reflection'] as String? ?? '',
          createdAt: reflectionDate,
        ));
      }
    }

    final pods = await FirestoreService.instance.getAllPods(user.uid);
    if (pods.success) {
      final podList = pods['pods'] as List? ?? [];
      for (final raw in podList) {
        final pod = Map<String, dynamic>.from(raw as Map);
        final reflection = pod['reflection'] as String? ?? '';
        final startDate = pod['startDate'] as String? ?? '';
        if (reflection.isEmpty || startDate.isEmpty) continue;
        if (all.any((r) => r.date == startDate)) continue;
        DateTime reflectionDate;
        try {
          final p = startDate.split('-');
          reflectionDate = DateTime(int.parse(p[0]), int.parse(p[1]), int.parse(p[2]));
        } catch (_) {
          reflectionDate = pod['createdAt'] is DateTime ? pod['createdAt'] as DateTime : DateTime.now();
        }
        all.add(_ReflectionItem(
          id: pod['id'] as String? ?? startDate,
          date: startDate,
          dateObj: reflectionDate,
          reflection: reflection,
          createdAt: reflectionDate,
        ));
      }
    }

    final current = await FirestoreService.instance.getPodReflection(user.uid);
    if (current.success && (current.reflection ?? '').isNotEmpty) {
      final todayId = getDateId();
      if (!all.any((r) => r.date == todayId)) {
        all.add(_ReflectionItem(
          id: 'current',
          date: current['dateId'] as String? ?? todayId,
          dateObj: DateTime.now(),
          reflection: current.reflection ?? '',
          createdAt: current['createdAt'] is DateTime ? current['createdAt'] as DateTime : DateTime.now(),
        ));
      }
    }

    all.sort((a, b) => (b.createdAt ?? DateTime(0)).compareTo(a.createdAt ?? DateTime(0)));
    setState(() {
      _reflections = all;
      _filtered = all;
      _reflectionDays = all.map((r) => CalendarDayMarker(date: r.date)).toList();
      _loading = false;
    });
    _applyFilter();
  }

  void _applyFilter() {
    if (_selectedDate == null) {
      setState(() => _filtered = _reflections);
      return;
    }
    final id = getDateId(_selectedDate);
    setState(() => _filtered = _reflections.where((r) => r.date == id).toList());
  }

  int _currentIndex() {
    if (_selectedDate == null) return -1;
    return _reflections.indexWhere((r) => r.date == getDateId(_selectedDate));
  }

  @override
  Widget build(BuildContext context) {
    final isDark = context.watch<ThemeNotifier>().isDarkMode;

    return Scaffold(
      backgroundColor: HubTheme.scaffoldBg(isDark),
      body: Stack(
        children: [
          SafeArea(
            child: SingleChildScrollView(
              padding: const EdgeInsets.fromLTRB(24, 16, 24, 32),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Row(
                    children: [
                      IconButton(
                        icon: Icon(LucideIcons.arrowLeft, color: isDark ? Colors.white : Colors.grey.shade800),
                        onPressed: () => context.go(AppRoutes.pod),
                      ),
                      Icon(LucideIcons.sparkles, color: isDark ? const Color(0xFFFDD663) : const Color(0xFFE6B3BA)),
                      const SizedBox(width: 8),
                      Text('Reflections', style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold, color: isDark ? Colors.white : Colors.grey.shade800)),
                    ],
                  ),
                  Padding(
                    padding: const EdgeInsets.only(left: 56, bottom: 12),
                    child: Text('All your crew reflections', style: TextStyle(color: isDark ? Colors.grey.shade400 : Colors.grey.shade600, fontSize: 14)),
                  ),
                  Padding(
                    padding: const EdgeInsets.only(left: 56, bottom: 16),
                    child: InkWell(
                      onTap: () => setState(() => _calendarOpen = true),
                      child: Container(
                        padding: const EdgeInsets.all(12),
                        decoration: BoxDecoration(
                          color: isDark ? const Color(0xFF262626) : Colors.white,
                          borderRadius: BorderRadius.circular(8),
                        ),
                        child: Row(
                          children: [
                            Icon(LucideIcons.calendar, color: isDark ? const Color(0xFF7DD3C0) : const Color(0xFF87A96B), size: 18),
                            const SizedBox(width: 12),
                            Expanded(
                              child: Column(
                                crossAxisAlignment: CrossAxisAlignment.start,
                                children: [
                                  Text(
                                    _selectedDate != null ? formatDateForDisplay(_selectedDate) : 'Search by date',
                                    style: TextStyle(fontWeight: FontWeight.w500, color: isDark ? Colors.white : Colors.grey.shade800),
                                  ),
                                  Text(_selectedDate != null ? 'Tap to change date' : 'Tap to select a date', style: TextStyle(fontSize: 12, color: Colors.grey)),
                                ],
                              ),
                            ),
                            if (_selectedDate != null)
                              IconButton(
                                icon: const Icon(LucideIcons.x, size: 18),
                                onPressed: () {
                                  setState(() => _selectedDate = null);
                                  _applyFilter();
                                },
                              ),
                          ],
                        ),
                      ),
                    ),
                  ),
                  if (_selectedDate != null && _filtered.isNotEmpty)
                    Row(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        IconButton(
                          onPressed: _currentIndex() > 0
                              ? () {
                                  final prev = _reflections[_currentIndex() - 1];
                                  setState(() => _selectedDate = prev.dateObj);
                                  _applyFilter();
                                }
                              : null,
                          icon: const Icon(LucideIcons.chevronLeft),
                        ),
                        Text(formatDateForDisplay(_selectedDate), style: TextStyle(fontWeight: FontWeight.w600, color: isDark ? Colors.white : Colors.grey.shade800)),
                        IconButton(
                          onPressed: _currentIndex() >= 0 && _currentIndex() < _reflections.length - 1
                              ? () {
                                  final next = _reflections[_currentIndex() + 1];
                                  setState(() => _selectedDate = next.dateObj);
                                  _applyFilter();
                                }
                              : null,
                          icon: const Icon(LucideIcons.chevronRight),
                        ),
                      ],
                    ),
                  if (_loading)
                    const ListSkeleton(count: 5)
                  else if (_filtered.isEmpty)
                    Padding(
                      padding: const EdgeInsets.all(32),
                      child: Center(
                        child: Text(
                          _selectedDate != null ? 'No reflection found for the selected date.' : 'No reflections yet. Start chatting in your crew to generate reflections!',
                          textAlign: TextAlign.center,
                          style: TextStyle(color: isDark ? Colors.grey.shade400 : Colors.grey.shade600),
                        ),
                      ),
                    )
                  else
                    ..._filtered.map((r) => Container(
                          margin: const EdgeInsets.only(bottom: 12),
                          padding: const EdgeInsets.all(16),
                          decoration: BoxDecoration(
                            color: isDark ? const Color(0xFF262626) : Colors.white,
                            borderRadius: BorderRadius.circular(16),
                          ),
                          child: Row(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              Icon(LucideIcons.check, color: isDark ? const Color(0xFF81C995) : const Color(0xFF87A96B), size: 18),
                              const SizedBox(width: 12),
                              Expanded(
                                child: Column(
                                  crossAxisAlignment: CrossAxisAlignment.start,
                                  children: [
                                    Text(formatReflectionDateTime(r.createdAt, dateFallback: r.date), style: TextStyle(fontSize: 12, color: Colors.grey)),
                                    const SizedBox(height: 8),
                                    Text(r.reflection, style: TextStyle(fontSize: 14, height: 1.5, color: isDark ? Colors.grey.shade300 : Colors.grey.shade700)),
                                  ],
                                ),
                              ),
                            ],
                          ),
                        )),
                ],
              ),
            ),
          ),
          CalendarPopup(
            isOpen: _calendarOpen,
            onClose: () => setState(() => _calendarOpen = false),
            selectedDate: _selectedDate ?? DateTime.now(),
            chatDays: _reflectionDays,
            onDateSelect: (d) {
              setState(() => _selectedDate = d);
              _applyFilter();
            },
          ),
        ],
      ),
    );
  }
}
